dev/src/reference.ts

```diff
@@ -2020,26 +2017,30 @@ export class Query<T = DocumentData> {
       callback();
     });
 
-    this.firestore.initializeIfNeeded(tag).then(() => {
-      this._firestore
-        .requestStream('runQuery', request, tag)
-        .then(backendStream => {
-          backendStream.on('error', err => {
-            logger(
-              'Query._stream',
-              tag,
-              'Query failed with stream error:',
-              err
-            );
+    this.firestore
+      .initializeIfNeeded(tag)
+      .then(async () => {
```

Comment:
`async` is not needed, there is no `await` inside.

Reply:
I added a comment. `async` makes sure that any exception from `this.toProto` is converted into a rejected Promise.